### DISCUSSION TITLE
Does @nuxt/apollo need graphql-tag or not?

### DISCUSSION THREAD

**Discussion by mitya33** [nuxt/nuxt#9504] [Questions (Nuxt 2)]
I'm trying to establish whether using Apollo with Nuxt requires the `graphql-tag` or not. Most tutorials I've followed install it, and the docs [say](https://www.npmjs.com/package/@nuxtjs/apollo#3--loading-gql-or-graphql-files-optional) it's "optional" if you want to load `.gql` files.

Yet the [official Apollo demo for Nuxt](https://github.com/nuxt/nuxt.js/tree/dev/examples/vue-apollo) shows Apollo being used with separate `.gql` files _without_ having `graphql-tag` installed. (Or is the assumption that, to run it, you'd also install `graphql-tag`?)

So what precisely does `graphql-tag` do and is or is it not needed to use Apollo with Nuxt?

Thanks!

**Comment by TheAlexLichter** [comment]
If you take a look at the [tests](https://github.com/nuxt-community/apollo-module/blob/master/test/fixture-local-state/pages/index.vue) of the module, it shows that using `graphql-tag` is necessary if you want to **inline** queries. If you extract them into own `.gql` files (e.g. in [this test](https://github.com/nuxt-community/apollo-module/blob/master/test/fixture/pages/index.vue)) and import them, the responsible webpack handler should correctly transform these and `graphql-tag` is not necessary.

**Reply by mitya33** [reply]
How interesting, thank you. The YT tutorial I followed said the opposite, that they were importing graphql-tag precisely _in order to_ be able to use separated .gql files. Minefield!
